### PR TITLE
Refactor context

### PR DESCRIPTION
**Pull request by ching56**
1. Abstract the create context code into `createContext` and `createPersistContext`
2. export the data hook directly. Originally when we want to use the context data, we need to import  the useContext and the context, now we can use data directly

### CODE REVIEW

**Comment by TseHang** [comment]
LGTM!

Especially great idea that a syntactic sugar like `useUser`.

File: src/utils/context.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      storageKey,
+      createDefaultData,
+      {
+        persist: props.persist,
+        version,
+      },
+    );
+
+    return (
+      <Context.Provider value={{ data, setData, restored }}>
```
Comment (TseHang) [inline src/utils/context.tsx]: I have a new idea that
how about we return all format like `[data, setData, restored]`.

Cause now our format becomes more fixed, and the advantage is we don't need to change name every time when we get its reference.
Let user can set by his own. (like useState)

File: src/utils/context.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  };
+
+  const useData = () => {
+    const context = React.useContext(Context);
+    if (!context) {
+      throw new Error(
+        `Context error: context [${storageKey}] must be used within a Provider`,
+      );
+    }
+    const { data, setData, restored } = context;
```
Comment (TseHang) [inline src/utils/context.tsx]: If our context format is `[data, setData, restored]`.
I think we can directly return the whole context, cause it is clear.

And our use case may be like

 
`screen.tsx`
```javascript
import { useUser } from '@/contexts/userContext';
// ...
const [user, setUser, restored] = useUser(); 
// ...
```


also, if some body want to custom his hook, he can

`user.ts`
```
import { UserContext } from '@/contexts/userContext';
// ...
const [user, setUser, restored] = useContext(UserContext); 
const setUserById = () => ({ ... });
return {
  user,
  setUser,
  setUserById,
  ...
}
// ...
```

**Comment by TseHang** [comment]
Seems it is good feature if we directly put in [react-native-use-persist-storage](https://github.com/visuallylab/react-native-use-persist-storage#readme)

File: src/utils/context.tsx (hunk 1 above)
Comment (ching56) [reply src/utils/context.tsx]: Sure, it is great, let me update it

**Comment by ching56** [comment]
Updated, please review again

**Pull request merged**